### PR TITLE
add pagination

### PR DESCRIPTION
**Pull request by DidenkoDenis306**
[DEMO LINK](https://DidenkoDenis306.github.io/react_pagination/)

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -1,18 +1,40 @@
...
 import { getNumbers } from './utils';
+import { Pagination } from './components/Pagination/Pagination';
 
 // eslint-disable-next-line @typescript-eslint/no-unused-vars
 const items = getNumbers(1, 42)
   .map(n => `Item ${n}`);
 
 export const App: React.FC = () => {
+  const [currentPage, setCurrentPage] = useState(1);
+  const [perPage, setPerPage] = useState(5);
```
Comment (33Orange) [inline src/App.tsx:12]: Use more descriptive names
```suggestion
  const [limitPerPage, setLimitPerPage] = useState(5);
```

File: src/components/Pagination/Pagination.tsx [hunk 2]
```diff
@@ -1 +1,116 @@
...
+          </a>
+        </li>
+      </ul>
+      <ul>
+        {validVisibleItems.map(item => (
+          <li data-cy="item" key={item}>
+            {item}
+          </li>
+        ))}
+      </ul>
```
Comment (33Orange) [inline src/components/Pagination/Pagination.tsx:113]: Component `pagination` should render only pagination panel without items

File: src/components/Pagination/Pagination.tsx [hunk 3]
```diff
@@ -1 +1,116 @@
...
+
+export const Pagination: React.FC<Props> = ({
+  total,
+  perPage,
+  currentPage,
+  onPageChange,
+}) => {
+  const pagesAmount = Math.ceil(total / perPage);
+
+  const visibleItems = [];
```
Comment (33Orange) [inline src/components/Pagination/Pagination.tsx:19]: Move list of items to app component and render list in app

File: src/components/Pagination/Pagination.tsx [hunk 4]
```diff
@@ -1 +1,116 @@
...
+  const visiblePages = [];
+
+  for (let i = 1; i <= pagesAmount; i += 1) {
+    visiblePages.push(i);
+  }
+
+  for (let i = 1; i <= perPage; i += 1) {
+    visibleItems.push(`Item ${i + perPage * (currentPage - 1)}`);
+  }
```
Comment (33Orange) [inline src/components/Pagination/Pagination.tsx:28]: Find first index and last index depending on the current page and use `slice` on your array of items

File: src/components/Pagination/Pagination.tsx [hunk 5]
```diff
@@ -1 +1,116 @@
...
+  onPageChange,
+}) => {
+  const pagesAmount = Math.ceil(total / perPage);
+
+  const visibleItems = [];
+  const visiblePages = [];
+
+  for (let i = 1; i <= pagesAmount; i += 1) {
+    visiblePages.push(i);
+  }
```
Comment (33Orange) [inline src/components/Pagination/Pagination.tsx:24]: use `getNumbers` function, which you used for creating array of numbers, instead

File: src/components/Pagination/Pagination.tsx [hunk 6]
```diff
@@ -1 +1,116 @@
...
+
+  return (
+    <>
+      <ul className="pagination">
+        <li className={cn(
+          'page-item',
+          {
+            disabled: isSelectedPageFirst,
+          },
+        )}
```
Comment (33Orange) [inline src/components/Pagination/Pagination.tsx:56]: Remove unnecessary linebreaks

```suggestion
        <li className={cn('page-item', {
            disabled: isSelectedPageFirst,
          })}
```

File: src/components/Pagination/Pagination.tsx [hunk 7]
```diff
@@ -1 +1,116 @@
...
+          </a>
+        </li>
+        {visiblePages.map(page => (
+          <li
+            className={cn(
+              'page-item',
+              {
+                active: currentPage === page,
+              },
+            )}
```
Comment (33Orange) [inline src/components/Pagination/Pagination.tsx:75]: Same about linebreaks in classname

File: src/components/Pagination/Pagination.tsx [hunk 8]
```diff
@@ -1 +1,116 @@
...
+            </a>
+          </li>
+        ))}
+        <li
+          className={cn(
+            'page-item',
+            {
+              disabled: isSelectedPageLast,
+            },
+          )}
```
Comment (33Orange) [inline src/components/Pagination/Pagination.tsx:94]: Same about linebreaks in classname

**Review by 33Orange — CHANGES_REQUESTED** [review]
Good job. Almost done

A few moments to fix and improve your code

File: src/App.tsx [hunk 9]
```diff
@@ -1,18 +1,42 @@
...
   .map(n => `Item ${n}`);
 
 export const App: React.FC = () => {
+  const [currentPage, setCurrentPage] = useState(1);
+  const [limitPerPage, setLimitPerPage] = useState(5);
+
+  const amountOfItems = items.length;
+
+  const firstItem = limitPerPage * (currentPage - 1);
+  let lastItem = limitPerPage * currentPage;
```
Comment (33Orange) [inline src/App.tsx:17]:
```suggestion
  const lastItem = limitPerPage * currentPage > amountOfItems ? amountOfItems  : limitPerPage * currentPage;
```

File: src/App.tsx [hunk 10]
```diff
@@ -1,18 +1,42 @@
...
+
+  const currentInfo = () => {
+    if (lastItem > amountOfItems) {
+      lastItem = amountOfItems;
+    }
```
Comment (33Orange) [inline src/App.tsx:22]:
```suggestion
```

File: src/App.tsx [hunk 11]
```diff
@@ -1,18 +1,42 @@
...
 const items = getNumbers(1, 42)
   .map(n => `Item ${n}`);
 
 export const App: React.FC = () => {
+  const [currentPage, setCurrentPage] = useState(1);
+  const [limitPerPage, setLimitPerPage] = useState(5);
+
+  const amountOfItems = items.length;
+
+  const firstItem = limitPerPage * (currentPage - 1);
```
Comment (33Orange) [inline src/App.tsx:16]: 5 * 1 - 5 = 0
5 * 2 - 5 = 5
5 * 3 - 5 = 10

```suggestion
  const firstItem = limitPerPage * currentPage - limitPerPage;
```

File: src/App.tsx [hunk 12]
```diff
@@ -1,18 +1,42 @@
...
+  let lastItem = limitPerPage * currentPage;
+
+  const currentInfo = () => {
+    if (lastItem > amountOfItems) {
+      lastItem = amountOfItems;
+    }
+
+    return `Page ${currentPage} (items ${firstItem} - ${lastItem} of ${amountOfItems})`;
```
Comment (33Orange) [inline src/App.tsx:24]:
```suggestion
    return `Page ${currentPage} (items ${firstItem + 1} - ${lastItem} of ${amountOfItems})`;
```

**Review by 33Orange — CHANGES_REQUESTED** [review]
Almost done

Create a variable for `limitPerPage * currentPage`

**Review by And678 — APPROVED** [review]
Good work, approve :heavy_check_mark:

**Pull request closed without merging**